src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -205,13 +205,16 @@ public void StartConnectAsync()
 
             this.OnStartConnect();
 
+            if (this.MaxOutboundConnections == 0)
```

Comment:
@noescape00 The issue putting the MaxOutboundConnections outside the async loop is that we can add nodes to PeerConnectorAddNode's collection via an RPC call. So if we do this, this connector never starts up and as such when we call that RPC method, it won't have any affect. Remember that PeerConnectorAddNode must always run.

Reply:
Good notice. Didn't know that we can use addNode during the runtime. 
Fixed.